### PR TITLE
Chess Hook

### PR DESCRIPTION
**Pull request by namin**
As promised, a chess hook on top of python-chess
https://python-chess.readthedocs.io/en/latest/

- shallow: does not wrap python-chess

- tiny: only provides GameState for now, which is enough to run the
  alpha beta agent

- incomplete: it would be interesting to support other parts of the book...

Suggestions welcome... might not be worth merging as is. In addition, this creates a dependency to python-chess, so maybe not ideal to include in the dlgo lib.

Thanks :)

### CODE REVIEW

File: code/dlgo/chess/chessboard.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    def apply_move(self, move):
+        """Return the new GameState after applying the move."""
+        next_board = self.board.copy()
+        next_board.push(move)
+        return GameState(next_board)
+
+    @classmethod
+    def new_game(cls):
+        board = _chess.Board()
```
Comment (maxpumperla) [inline code/dlgo/chess/chessboard.py]: I'm a bit confused about your `_chess` import. where do I find the code for `Board`, for instance?

**Comment by maxpumperla** [comment]
@namin super cool, thank you! just a question about the `chess` module.

File: code/dlgo/chess/chessboard.py (hunk 1 above)
Comment (namin) [reply code/dlgo/chess/chessboard.py]: That's the `python-chess` part. The code for board is here: https://python-chess.readthedocs.io/en/latest/core.html#chess.Board

**Comment by maxpumperla** [comment]
ah, sorry. it's based on python-chess, I see. can you please add a little readme telling us how to install dependencies and run your tool? that'd be really cool.

**Comment by maxpumperla** [comment]
perfect, thanks

**Comment by namin** [comment]
Cool, thanks. My pleasure :)

**Pull request merged**